beta-src/src/models/PathClass.ts

```diff
@@ -0,0 +1,337 @@
+              (node.hasPath() && nextNodeRank < node.getMaxRank())
+            );
+          },
+          (node) => {
+            /* additional condition an alternative route nodes have to fullfill
+             * for optimization:
+             *
+             * node not already used in search of same rank or lower before
+             *		(=> if it is, there are no chances of finding an alternative route)
+             */
+            return nextNodeRank < node.getMaxRank();
+          },
+        );
+
+        if (alternativeRouteSearch.findPath()) {
+          // there exists an alternativeRoute
+          let alternativeRoute = alternativeRouteSearch.path;
+
+          // first check if alternativeRoute reaches endNode or path with lower rank
+          if (!fEndNode(alternativeRoute.node)) {
+            // not EndNode -> reconnect paths
+            alternativeRoute = alternativeRoute.attachToNodePath();
+          }
+
+          // append alternativeRoute to current path
+          alternativeRoute.removeFirst().newPathToNode(this);
+
+          // next set the route complete
+          alternativeRoute.setComplete();
+
+          // now search alternativeRoute for alternativeRoutes (beginning at the node next to end node)
+          alternativeRoute.pathToNode.searchAlternativeRoutes(fEndNode);
+          /*
+           * Note, that not extra check is needed for pathToNode:
+           * Even if the alternativeRoute heads directly into endNode,
+           * pathToNode is at least this. So the extreme case is that other
+           * paths from this node are checked (which is wanted)
+           */
+        } else {
+          // all alternative routes from this node are found
+          this.alternativeChecked = true;
+        }
+      }
+    }
+
+    if (this.pathToNode !== null)
+      this.pathToNode.searchAlternativeRoutes(fEndNode);
+  }
+
+  /*
+   * Returns an alternative path that does not pass through this.node.
+   * Alternative routes have to be searched before-
+   */
+  getAlternativeRoute() {
+    if (!this.alternativeChecked) return null;
+
+    if (this.pathToNode === null) return null;
+
+    if (this.pathToNode.pathNextNodes.length > 1) {
+      // there is a fork in the path at node before
+      // find a path to next node of previous node that is not this.
+      // and return the complete path
+      return this.pathToNode.pathNextNodes
+        .find((path) => {
+          return path !== this;
+        }, this)
+        .getLastPathNode();
+    }
+    return this.pathToNode.getAlternativeRoute();
+  }
+
+  /*
+   * Checks if this path can be appended to path so a simple path is preserved.
+   * So this basically checks if both paths share any common nodes apart from
+   * path.lastNode and this.firstNode.
+   *
+   * Implemented recursivly
+   */
+  canBeAppendedTo(path) {
+    path.markPath(path);
+
+    let retValue = true;
+
+    if (this.node.inPath === path) retValue = false;
```

Comment:
braces on all ifs please. one line ifs are bad, especially in this case where we have code that is following it.